File: Platforms/Anf.Engine.Easy/Visiting/PageBox.cs
using System;
using System.Threading.Tasks;

namespace Anf.Easy.Visiting
{
    public struct PageBox<TResource> : IComicVisitPage<TResource>,IDisposable
    {
        public bool DoNotDispose { get; set; }

        public ComicPage Page { get; set; }

        public TResource Resource { get; set; }

        public void Dispose()
        {
            if (!DoNotDispose&& Resource is IDisposable disposable)
            {
                disposable.Dispose();
            }
        }
    }
}

File: anf-rs/crates/anf-easy/src/visiting/page_box.rs
use anf_core::ComicPage;

/// PageBox: wraps a page with its loaded resource
/// Matches C# PageBox<TResource> : IComicVisitPage<TResource>
#[derive(Debug, Clone)]
pub struct PageBox<T> {
    pub page: ComicPage,
    pub resource: Option<T>,
}
